daemon/cmd/daemon.go

```diff
@@ -939,14 +939,16 @@ func NewDaemon(ctx context.Context, cleaner *daemonCleanup,
 		}
 		// ipt.InstallRules() (called by Reinitialize()) happens later than
 		// this  statement, so it's OK to fallback to iptables-based MASQ.
-		option.Config.EnableBPFMasquerade = false
-		log.Warn(msg + " Falling back to iptables-based masquerading.")
-		// Too bad, if we need to revert to iptables-based MASQ, we also cannot
-		// use BPF host routing since we need the upper stack.
-		if !option.Config.EnableHostLegacyRouting {
-			option.Config.EnableHostLegacyRouting = true
-			log.Infof("BPF masquerade could not be enabled. Falling back to legacy host routing (--%s=\"true\").",
-				option.EnableHostLegacyRouting)
+		if len(msg) > 0 {
```

Comment:
The check on `msg` belongs in the refactoring.

Reply:
ack